packages/server/src/model-view.ts

```diff
@@ -30,8 +31,18 @@ export function view<T>(
       return getModelState(model)
     }
     const newState = loadInput(getter)
-    state = newState
     isValid = true
+    if (opts?.compare) {
+      if (opts.compare === true && newState === state) return newState
+      if (
+        typeof opts.compare === 'function' &&
+        state !== undefined &&
```

Comment:
What is `state` is null? Is this possible? Do we want to run comparison function in that case? 
Also, is it possible that both `state` and `newState` are undefined? What should happen in such case?

Reply:
state may be undefined if the previous state was empty because the dependencies were still loading. So probably the view is getting re-evaluated with the real state at this point